sklearn/utils/estimator_checks.py

```diff
@@ -2369,6 +2375,32 @@ def check_outliers_train(name, estimator_orig, readonly_memmap=True):
             check_outlier_corruption(num_outliers, expected_outliers, decision)
 
 
+def check_outlier_contamination(name, estimator_orig):
+    # Check that the contamination parameter is in (0.0, 0.5] when it is an
+    # interval constraint.
+
+    if not hasattr(estimator_orig, "_parameter_constraints"):
+        # Only estimator implementing parameter constraints will be checked
+        return
+
+    if "contamination" not in estimator_orig._parameter_constraints:
+        return
+
+    contamination_constraints = estimator_orig._parameter_constraints["contamination"]
+    if not any([isinstance(c, Interval) for c in contamination_constraints]):
+        raise AssertionError(
+            "contamination constraints should contain a Real Interval constraint."
+        )
+
+    for constraint in contamination_constraints:
+        if isinstance(constraint, Interval):
+            validity_interval = Interval(Real, 0, 0.5, closed="right")
+            assert (
+                constraint.left in validity_interval
+                and constraint.right in validity_interval
```

Comment:
Sorry, my suggestion did not work. This one should work, or you can revert to the previous one if you prefer
```suggestion
            assert (
                constraint.left >= 0 and 0 not in constraint and constraint.right <= 0.5
```